### PR TITLE
feature(pdd): enable P/D disaggregation with NIXL host KV transfer

### PR DESCRIPTION
**Pull request by rebel-ykchoi**
### 🚀 Summary of Changes

wire vLLM KV transfer to a RBLN-specific NIXL connector and host-side buffers so prefill/decode can run on separate engines with H2H transfer.

KV connector / registration
- add RblnNixlConnector (scheduler/worker) extending upstream NixlConnector:
- register connector name "RblnNixlConnector" in kv_connector factory.

Platform
- expose NIXL hints: get_nixl_supported_devices (rbln -> cpu) and get_nixl_memory_type ("DRAM").

Scheduler (rbln_scheduler.py)
- handle kv_consumer request to be scheduled with other requests in decode stage

Model runner (rbln_model_runner.py)
- override maybe_get_kv_connector_output(..., wait_for_save) using last prefill chunk.
- replace generic copy_kv_blocks with rbln_copy_kv_blocks using runtime _update_kv_cache / _fetch_kv_cache
- bind_kv_cache_name + per-layer names for mark_static_address when compiling.

Attention backend (flash_attention.py)
- report backend name as FLASH_ATTN for upstream compatibility.

Tests
- tests/torch_compile/e2e/v1/kv_connector/nixl_integration/run_accuracy_test.sh

> *What does this PR do? What feature, fix, or improvement does it bring?*

---

### 📌 Related Issues / Tickets

* Resolves #
* Related to #

---

### ✅ Type of Change

* [ ] 🚀 Release (`release`)
* [x] ✨ Feature (`feature`)
* [ ] 🧠 Model support (`model`)
* [ ] 🧬 Core engine changes (`core`)
* [ ] 🛠 Bug fix (`fix`)
* [ ] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [ ] ❓ Other (`other`): please describe

---

### 🧪 How to Test

1. Run `...`
2. Verify output: `...`
3. Edge case tested: `...`

---

### 📸 Screenshots / Logs (if applicable)

---

### 📋 Checklist

* [ ] PR title follows Conventional Commits format
* [ ] This PR is linked to an existing issue
* [ ] The test method is described, and the expected result is clearly stated
* [ ] Relevant documentation has been updated (if applicable)

---

### 💬 Notes

---

### CODE REVIEW

File: vllm_rbln/v1/worker/rbln_worker.py [hunk 1]
```diff
@@ -145,10 +150,6 @@ def wake_up(self, tags: list[str] | None = None) -> None:
         logger.warning("sleep mode is not supported on RBLN, ignore it.")
         pass
```
[the remark this answers is not in the document]
Comment (rebel-ykchoi) [reply vllm_rbln/v1/worker/rbln_worker.py:152]: fixed unit tests.

File: vllm_rbln/v1/worker/rbln_model_runner.py [hunk 2]
```diff
@@ -4440,7 +4474,38 @@ def initialize_kv_cache(self, kv_cache_config: KVCacheConfig) -> None:
...
+                    kv_caches = src_kv_caches
+                    copy_fn = runtime._update_kv_cache
+                else:
+                    kv_caches = dst_kv_caches
+                    copy_fn = runtime._fetch_kv_cache
+
+                for idx in src_block_ids:
+                    for kv_name, kv_cache in kv_caches.items():
+                        block_size = kv_cache.shape[-2]
+                        copy_fn(kv_cache.data_ptr(), idx, 0, block_size, kv_name)
```
[the remark this answers is not in the document]
Comment (rebel-ykchoi) [reply vllm_rbln/v1/worker/rbln_model_runner.py:4509]: added an assertion to guarantee the equality of the two lists.

File: tests/torch_compile/e2e/v1/kv_connector/nixl_integration/toy_proxy_server.py [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+        req_data["max_completion_tokens"] = 1
+    if "stream_options" in req_data:
+        del req_data["stream_options"]
+    headers = {
+        "Authorization": f"Bearer {os.environ.get('OPENAI_API_KEY')}",
+        "X-Request-Id": request_id,
+    }
+
+    response = await client_info["client"].post(
+        endpoint, json=req_data, headers=headers
```
[the remark this answers is not in the document]
Comment (rebel-ykchoi) [reply tests/torch_compile/e2e/v1/kv_connector/nixl_integration/toy_proxy_server.py:196]: removed the leading slash from the endpoint

**Comment by rebel-jaehwang** [comment]
added a unit test for the new scheduler behavior

File: vllm_rbln/v1/worker/utils.py [hunk 4]
```diff
@@ -422,3 +425,53 @@ def set_omp_num_threads(
...
+
+
+def bind_kv_cache_name(
+    kv_caches: dict[str, torch.Tensor],
+    runner_kv_cache_names: list[str],
+    num_attn_module: int = 1,
+) -> None:
+    """
+    Bind the allocated KV cache name to ModelRunner and forward context so
+    that the KV cache can be used in the forward pass.
```
Comment (rebel-jaehwang) [inline vllm_rbln/v1/worker/utils.py]: weird name and description of the function. it's just returning the kv cache names. 
```suggestion
def get_kv_cache_names(
    kv_caches: dict[str, torch.Tensor],
    num_attn_module: int = 1,
) -> list[str]:
    """
    Returns the names of the KV caches.
```

Comment (rebel-ykchoi) [reply vllm_rbln/v1/worker/utils.py]: fixed it.

**Review by rebel-jaehwang — APPROVED** [review]

**Comment by rebel-ykchoi** [comment]
> added a unit test for the new scheduler behavior

Thanks, I have organized the unit tests, incorporating the test case you provided.
